cocotb/share/lib/vhpi/VhpiImpl.cpp

```diff
@@ -92,13 +92,23 @@ void VhpiImpl::get_sim_time(uint32_t *high, uint32_t *low)
     *low = vhpi_time_s.low;
 }
 
+template <typename T>
+int log10int(T v)
+{
+    int i = -1;
+    do {
+        v /= 10;
+        i += 1;
+    } while (v);
+    return i;
+}
+
 void VhpiImpl::get_sim_precision(int32_t *precision)
 {
     /* The value returned is in number of femtoseconds */
     vhpiPhysT prec = vhpi_get_phys(vhpiResolutionLimitP, NULL);
     uint64_t femtoseconds = ((uint64_t)prec.high << 32) | prec.low;
-    double base = 1e-15 * femtoseconds;
-    *precision = (int32_t)log10(base);
+    *precision = log10int(base) - 15;
```

Comment:
Where did `base` go?

Reply:
```suggestion
    *precision = log10int(femptoseconds) - 15;
```

Reply:
Whoops... Seems I'll need to rebase #1329 on top of this to test this interface.